browser/plugins/test/network-breadcrumbs.test.js

```diff
@@ -0,0 +1,171 @@
+        request.open('GET', '/')
+
+        request.addEventListener('load', () => {
+          expect(client.breadcrumbs.length).toBe(1)
+          done()
+        })
+
+        request.send()
+      })
+    })
+
+    it('should leave a breadcrumb when an XMLHTTPRequest has a failed response', (done) => {
+      setup((client) => {
+        const request = new XMLHttpRequest()
+        request.open('GET', '/this-does-not-exist')
+
+        request.addEventListener('load', () => {
+          expect(client.breadcrumbs.length).toBe(1)
+          expect(client.breadcrumbs[0]).toEqual(jasmine.objectContaining({
+            type: 'network',
+            name: 'XMLHttpRequest failed',
+            metaData: {
+              status: 404,
+              request: 'GET /this-does-not-exist'
+            }
+          }))
+          done()
+        })
+
+        request.send()
+      })
+    })
+
+    // Browsers that support XDomainRequest will throw a real error when trying to do a cross
+    // domain request. We don't need a breadcrumb for this since bugsnag will track that error
+    if (!XDomainRequest) {
+      it('should leave a breadcrumb when an XMLHTTPRequest has a network error', (done) => {
+        setup((client) => {
+          const request = new XMLHttpRequest()
+          request.open('GET', 'https://api.bugsnag.com')
+
+          request.addEventListener('error', () => {
+            expect(client.breadcrumbs.length).toBe(1)
+            expect(client.breadcrumbs[0]).toEqual(jasmine.objectContaining({
+              type: 'network',
+              name: 'XMLHttpRequest error',
+              metaData: {
+                request: 'GET https://api.bugsnag.com'
+              }
+            }))
+            done()
+          })
+
+          request.send()
+        })
+      })
+
+      it('should not leave a breadcrumb for request to bugsnag notify endpoint', (done) => {
+        setup((client) => {
+          const request = new XMLHttpRequest()
+          request.open('GET', VALID_NOTIFIER.url)
+
+          request.addEventListener('error', () => {
+            expect(client.breadcrumbs.length).toBe(0)
+            done()
+          })
+
+          request.send()
+        })
+      })
+    }
+  }
+
+  if (global.fetch) {
+    it('should leave a breadcrumb when a fetch() resolves', (done) => {
+      setup((client) => {
+        global.fetch('/').then(() => {
+          expect(client.breadcrumbs.length).toBe(1)
+          expect(client.breadcrumbs[0]).toEqual(jasmine.objectContaining({
+            type: 'network',
+            name: 'fetch() succeeded',
+            metaData: {
+              status: 200,
+              request: 'GET /'
+            }
+          }))
+          done()
+        })
+      })
+    })
+    it('should leave a breadcrumb when a fetch() has a failed response', (done) => {
+      setup((client) => {
+        global.fetch('http://jsonplaceholder.typicode.com/posts/asdf').then(() => {
```

Comment:
does this test depend on http://jsonplaceholder.typicode.com/posts/asdf being available?

Reply:
This test assumes that that endpoint will be reachable and will return a 404.